rules/framework.bzl

```diff
@@ -522,12 +522,22 @@ def _copy_swiftmodule(ctx, framework_files):
     # only add a swift module to the SwiftInfo if we've actually got a swiftmodule
     swiftmodule_name = paths.split_extension(inputs.swiftmodule.basename)[0]
 
+    # find the first swift symbol_graph in the direct_modules of the deps of this target and forward it.
+    symbol_graph = None
+    for dep in ctx.attr.deps:
+        if SwiftInfo in dep and dep[SwiftInfo].direct_modules:
+            module = dep[SwiftInfo].direct_modules[0]
+            if module.swift and module.swift.symbol_graph:
+                symbol_graph = module.swift.symbol_graph
```

Comment:
Is there a reason that the symbol graph shouldn't be part of `inputs` and `outputs` / adjacent to the other components? The existing compiler artifacts are collected onto these struct(s) / follow this pattern and would suggest to keep the `.symbolgraph` similar to how we process `.swiftmodule` output

Reply:
The symbol graphs are added as extra outputs. They don't seem to be available in the rule last I checked

Reply:
Is there a way to access extra outputs from the rule? What you suggest is what I tried originally but the symbol graph wasn't in the list the rule gets. 

If there's a way to query the extra output groups though that would work

Reply:
Updated this to use `dep.output_groups`, in there it will have a `swift_symbol_graph` if the `--features=swift.emit_symbol_graph` is provided (which the .docc rule transitions on)